### ISSUE TITLE
Change Request: no-restricted-imports import/export configuration

### ISSUE THREAD

**Issue by JasonGore** [eslint/eslint#17395]
### ESLint version

8.36.0

### What problem do you want to solve?

`no-restricted-imports` currently applies configuration to both imports and exports. I just had a use case where I only wanted to ban exports of certain modules yet continue to allow them to be imported and used. I also reviewed `no-restricted-exports` but that only allows banning of exports by name and not by module.

### What do you think is the correct solution?

I had to fork `no-restricted-imports` and remove `ImportDeclaration` on the visitor object. I think it would be helpful if `no-restricted-imports` supported configuration for checking imports, exports or both (default behavior.)

### Participation

- [X] I am willing to submit a pull request for this change.

**Comment by nzakas** [comment]
Sorry we lost track of this. There's not really enough information here to comment on your request. Please provide some example code and explain what the rule currently does and what you'd like it to do instead.

**Comment by JasonGore** [comment]
I want to allow imports of certain modules but disallow their export. The specific modules in my case are devDeps and minimizing their downstream exposure by limiting re-export (tree shaking allows us to ensure `import` cases are also removed and that can't be done when `export` is used.) The `no-restricted-imports` as it exists now behaves as:

```
import { Something } from 'disallowed-module'; // disallowed
export { Something } from 'disallowed-module'; // disallowed
```

While I only wanted to ban `export` while allowing `import`:

```
import { Something } from 'disallowed-module'; // allowed
export { Something } from 'disallowed-module'; // disallowed
```

I thought it would be nice if the rule could be configured to apply to only exports, only imports, or both (default).

**Comment by JasonGore** [comment]
In this case I effectively allowed imports by forking the rule and commenting out the `ImportDeclaration` node on the visitor object in the rule.

**Comment by nzakas** [comment]
Ah I see, thanks for explaining. I'm not 100% sure this is the right solution to your problem. For instance, what if someone does this?

```js
import { Something } from 'disallowed-module';
export { Something };
```

What is the expected behavior there?

**Comment by JasonGore** [comment]
I'm primarily using the rule to prevent the simple re-export of dev deps. This is a way to work around the rule, but that is not how devs re-export deps in our repo. There are other workarounds as well, and it's hard to get 100% coverage of a rule without doing runtime analysis.

**Comment by mdjermanovic** [comment]
I also think it doesn't make sense to implement this in `no-restricted-imports`, because this rule restricts importing modules but not how they'll be used.

> I'm primarily using the rule to prevent the simple re-export of dev deps.

For this use case, you can probably use the [`no-restricted-syntax`](https://eslint.org/docs/latest/rules/no-restricted-syntax) rule. For example, this selector disallows re-exporting from `'disallowed-module'`:

```
:matches(ExportNamedDeclaration, ExportAllDeclaration)[source.value='disallowed-module']
```

```js
/*eslint no-restricted-syntax: ["error",
    {
        selector: ":matches(ExportNamedDeclaration, ExportAllDeclaration)[source.value='disallowed-module']",
        message: "Do not re-export devDeps."
    }
]*/

import { Something } from 'disallowed-module'; // allowed
export { Something } from 'disallowed-module'; // disallowed
export * from 'disallowed-module'; // disallowed
```

[Playground Demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tcmVzdHJpY3RlZC1zeW50YXg6IFtcImVycm9yXCIsXG4gICAge1xuICAgICAgICBzZWxlY3RvcjogXCI6bWF0Y2hlcyhFeHBvcnROYW1lZERlY2xhcmF0aW9uLCBFeHBvcnRBbGxEZWNsYXJhdGlvbilbc291cmNlLnZhbHVlPSdkaXNhbGxvd2VkLW1vZHVsZSddXCIsXG4gICAgICAgIG1lc3NhZ2U6IFwiRG8gbm90IHJlLWV4cG9ydCBkZXZEZXBzLlwiXG4gICAgfVxuXSovXG5cbmltcG9ydCB7IFNvbWV0aGluZyB9IGZyb20gJ2Rpc2FsbG93ZWQtbW9kdWxlJzsgLy8gYWxsb3dlZFxuZXhwb3J0IHsgU29tZXRoaW5nIH0gZnJvbSAnZGlzYWxsb3dlZC1tb2R1bGUnOyAvLyBkaXNhbGxvd2VkXG5leHBvcnQgKiBmcm9tICdkaXNhbGxvd2VkLW1vZHVsZSc7IC8vIGRpc2FsbG93ZWQiLCJvcHRpb25zIjp7ImVudiI6e30sInJ1bGVzIjp7fSwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9LCJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJtb2R1bGUifX19)

**Comment by JasonGore** [comment]
@mdjermanovic Will give this a try, thanks!
